Add GTI to Dataset

JouvinLea opened the pull request:
Start to implement the GTI in the Dataset Class.

registerrier commented on gammapy/spectrum/dataset.py:
```diff
@@ -805,6 +813,12 @@ def stack_edisp(self):
+        """Stack GTI
+        """
+        # TODO: use what Regis is implemented for GTI union
+        self.stack_gti = None
```
you should have typed `self.stacked_gti = None`.
Anyway I would `raise NotImplemented`

registerrier commented on gammapy/spectrum/extract.py:
```diff
@@ -134,6 +134,7 @@ def process(self, observation, bkg):
             backscale=1,
             backscale_off=bkg.a_off,
             obs_id=observation.obs_id,
+            gti=observation.gti,
```
This case requires a test. You could make sure that the correct GTI table is present in the existing tests.

JouvinLea commented:
@registerrier 
I added the method stacked_gti in the 'SpectrumDatasetOnOffStacker' and I added the test in test_dataset.
I also added a test in test_extract.

JouvinLea commented:
@adonath we can merge?

adonath commented:
Thanks @JouvinLea! I did some minor clean-up in a follow-up commit.
